Source/SteveSloth/Private/MyEnemyAttackingState.cpp

```diff
@@ -0,0 +1,69 @@
+/****************************************************************************************
+ * Copyright: SteveSloth
+ * Name: Ken Ferris
+ * Script: MyEnemyAttackingState.cpp
+ * Date: May 3, 2024
+ * Description: Enemy Attack State
+ * TODO:
+ * Known Bugs:
+ ****************************************************************************************/
+
+
+#include "MyEnemyAttackingState.h"
+#include "MyEnemyBaseClass.h"
+#include "Kismet/GameplayStatics.h"
+
+void UMyEnemyAttackingState::EnterState()
+{
+
+}
+
+void UMyEnemyAttackingState::ExitState()
+{
+
+}
+
+void UMyEnemyAttackingState::UpdateState(float deltaTime)
+{
+	if (IsPlayerInRange())
+	{
+		AttackPlayer();
+	}
+}
+
+bool UMyEnemyAttackingState::IsPlayerInRange() const
+{
+	AMyEnemyBaseClass* Owner = Cast<AMyEnemyBaseClass>(GetOuter());
+	if (!Owner)
+	{
+		UE_LOG(LogTemp, Error, TEXT("MyEnemyAttackingState::IsPlayerInRange - Owner is not valid"));
+		return false;
+	}
+
+	ACharacter* PlayerCharacter = UGameplayStatics::GetPlayerCharacter(GetWorld(), 0);
+	if (!PlayerCharacter)
+	{
+		UE_LOG(LogTemp, Error, TEXT("MyEnemyAttackingState::IsPlayerInRange - Player character not found"));
+		return false;
+	}
+
+	float Distance = FVector::Distance(Owner->GetActorLocation(), PlayerCharacter->GetActorLocation());
+	return Distance <= Owner->GetAttackRange();
+}
+
+void UMyEnemyAttackingState::AttackPlayer()
+{
+	AMyEnemyBaseClass* Owner = Cast<AMyEnemyBaseClass>(GetOuter());
+	if (!Owner)
+	{
+		UE_LOG(LogTemp, Error, TEXT("MyEnemyAttackingState::AttackPlayer - Owner is not valid"));
+		return;
+	}
+
+	ACharacter* PlayerCharacter = UGameplayStatics::GetPlayerCharacter(GetWorld(), 0);
```

Comment:
If you were listening in class this is not how we get access to the player.. Please follow what we did in class.. there is also some examples in chat in class.. and i shared my screen to show how as well..

add
ACharacter* Player; to the Header file

then add
Player = USteveSingleton::GetSteve()->GetPlayerCharacter(); to the Enter state of the cpp